homeassistant/components/light/tellduslive.py

```diff
@@ -0,0 +1,65 @@
+"""
+Support for Tellstick switches using Tellstick Net.
+
+This platform uses the Telldus Live online service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.tellduslive/
+
+"""
+import logging
+
+from homeassistant.components.tellduslive import (
+    TelldusLiveEntity,
+    TELLSTICK_TURNON,
+    TELLSTICK_TURNOFF,
+    TELLSTICK_DIM)
+from homeassistant.components.light import (
+    Light,
+    ATTR_BRIGHTNESS,
+    SUPPORT_BRIGHTNESS)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup lights."""
+    if discovery_info is None:
+        return
+    add_devices(TelldusLiveLight(hass, light) for light in discovery_info)
+
+
+class TelldusLiveLight(TelldusLiveEntity, Light):
+    """Representation of a light."""
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        try:
+            return int(self._device.get("statevalue"))
+        except ValueError:
+            return 255
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORT_BRIGHTNESS
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        # pylint: disable=attribute-defined-outside-init
+        self._brightness = self.brightness  # remember last set brightness
+        return (self._state == TELLSTICK_TURNON or
+                self._state == TELLSTICK_DIM)
+
+    def turn_on(self, **kwargs):
+        """Turn the light on."""
+        level = kwargs.get(ATTR_BRIGHTNESS, self._brightness)
+        if self._execute(TELLSTICK_DIM, level=level):
+            self._device['statevalue'] = level
+            self.update_ha_state()
+
+    def turn_off(self, **kwargs):
+        """Turn the light off."""
+        self._execute(TELLSTICK_TURNOFF)
```

Comment:
The state is updated in the parent class (`TelldusLiveDevice`), which also overrides `should_poll` to return `False`.